xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-api/src/main/java/org/xwiki/model/validation/edit/EditConfirmationScriptService.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.model.validation.edit;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.GroupBlock;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.stability.Unstable;
+import org.xwiki.text.XWikiToStringBuilder;
+
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage;
+
+/**
+ * This class provides the script services for handling document edit confirmation.
+ *
+ * @version $Id$
+ * @since 15.8RC1
+ */
+@Component
+@Singleton
+@Named("editConfirmation")
```

Comment:
I'm wondering if we want `$services.editConfirmation` or `$services.modelValidation.edit`?

Reply:
Hum, that's a good question. The think is, `ModelValidationScriptService` is in `xwiki-platform-model-validation-default` and `EditConfirmationScriptService` in `xwiki-platform-model-validation-api`.
So, if we name it `modelvalidation.edit`, we need to assume that `xwiki-platform-model-validation-default` is always installed (or that its replacement also provides a `modelvalidation` with a `get(String)` method).

Reply:
Hmm indeed that's an issue, and I imagine we cannot move ModelValidationScriptService to the API module?

Reply:
I see two options:
1. introduce a "second" `ModelValidationScriptService` in `xwiki-platform-model-validation-api` with a lower priority and that would just provide a `get(String)` method, and make the current `ModelValidationScriptService` inherit from it
2. move `ModelValidationScriptService` in `xwiki-platform-model-validation-api`. In this case we need to:
  - add a dependency to `xwiki-platform-security-authorization-api` which is fine
  - introduce an define an internal role `ReplaceCharacterEntityNameValidationConfiguration` in `xwiki-platform-model-validation-api`  and rename the current `ReplaceCharacterEntityNameValidationConfiguration` to `DefaultReplaceCharacterEntityNameValidationConfiguration` in `xwiki-platform-model-validation-api`, and make it inherit from the newly introduce role instead of using `roles = ReplaceCharacterEntityNameValidationConfiguration.class`
  
Feels like the second option is cleaner.

Reply:
Commit of option 2 https://github.com/xwiki/xwiki-platform/pull/2343/commits/cde389081427e60c7aebf64155ab8afe556ece5b

Reply:
> Feels like the second option is cleaner.

That's indeed cleaner, thanks!